agents/src/inference/stt.ts

```diff
@@ -0,0 +1,395 @@
+// SPDX-FileCopyrightText: 2025 LiveKit, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+import { type AudioFrame } from '@livekit/rtc-node';
+import { WebSocket } from 'ws';
+import { APIError, APIStatusError } from '../_exceptions.js';
+import { AudioByteStream } from '../audio.js';
+import { log } from '../log.js';
+import {
+  STT as BaseSTT,
+  SpeechStream as BaseSpeechStream,
+  type SpeechData,
+  type SpeechEvent,
+  SpeechEventType,
+} from '../stt/index.js';
+import { type APIConnectOptions, DEFAULT_API_CONNECT_OPTIONS } from '../types.js';
+import { type AudioBuffer, Event, Task, cancelAndWait, shortuuid, waitForAbort } from '../utils.js';
+import { connectWs, createAccessToken } from './utils.js';
+
+export type DeepgramModels =
+  | 'deepgram'
+  | 'deepgram/nova-3'
+  | 'deepgram/nova-3-general'
+  | 'deepgram/nova-3-medical'
+  | 'deepgram/nova-2'
+  | 'deepgram/nova-2-general'
+  | 'deepgram/nova-2-medical'
+  | 'deepgram/nova-2-phonecall';
+
+export type CartesiaModels = 'cartesia' | 'cartesia/ink-whisper';
+
+export type AssemblyaiModels = 'assemblyai';
+
+export interface CartesiaOptions {
+  min_volume?: number; // default: not specified
+  max_silence_duration_secs?: number; // default: not specified
+}
+
+export interface DeepgramOptions {
+  filler_words?: boolean; // default: true
+  interim_results?: boolean; // default: true
+  endpointing?: number; // default: 25 (ms)
+  punctuate?: boolean; // default: false
+  smart_format?: boolean;
+  keywords?: Array<[string, number]>;
+  keyterms?: string[];
+  profanity_filter?: boolean;
+  numerals?: boolean;
+  mip_opt_out?: boolean;
+}
+
+export interface AssemblyaiOptions {
+  format_turns?: boolean; // default: false
+  end_of_turn_confidence_threshold?: number; // default: 0.01
+  min_end_of_turn_silence_when_confident?: number; // default: 0
+  max_turn_silence?: number; // default: not specified
+  keyterms_prompt?: string[]; // default: not specified
+}
+
+export type STTModels = DeepgramModels | CartesiaModels | AssemblyaiModels;
```

Comment:
I think you need `| string` here or at consumption sites so people can adopt new models added to the service that aren't yet added to the type enums (same for the other types)